File: backend/app/constants/cinnamon_varieties.py
# Cinnamon variety constants extracted from yield dataset
# These should match the frontend constants

CINNAMON_VARIETIES = [
    {
        'value': 'Ceylon Cinnamon',
        'label': 'Ceylon Cinnamon',
        'description': 'Premium quality cinnamon, also known as "True Cinnamon"',
        'scientific_name': 'Cinnamomum verum',
        'characteristics': 'Sweet, delicate flavor with low coumarin content'
    },
    {
        'value': 'Alba',
        'label': 'Alba',
        'description': 'High-yielding variety with good commercial value',
        'scientific_name': 'Cinnamomum cassia var. Alba',
        'characteristics': 'Robust growth, medium flavor intensity'
    },
    {
        'value': 'Continental',
        'label': 'Continental',
        'description': 'Hardy variety suitable for various climatic conditions',
        'scientific_name': 'Cinnamomum cassia var. Continental',
        'characteristics': 'Strong flavor, good for spice production'
    }
]

# Extract just the values for easy usage
CINNAMON_VARIETY_VALUES = [variety['value'] for variety in CINNAMON_VARIETIES]

# Default variety (most common in dataset)
DEFAULT_CINNAMON_VARIETY = 'Ceylon Cinnamon'

def validate_cinnamon_variety(variety: str) -> bool:
    """Validate if the given variety is one of the supported types"""
    return variety in CINNAMON_VARIETY_VALUES

def get_variety_info(variety: str) -> dict:
    """Get detailed information about a specific variety"""
    for var in CINNAMON_VARIETIES:
        if var['value'] == variety:
            return var
    return None

File: mobile-app/constants/CinnamonVarieties.ts
// Cinnamon variety constants for Sri Lankan varieties
export const CINNAMON_VARIETIES = [
  {
    value: 'Sri Gemunu',
    label: 'Sri Gemunu', // Will be translated in UI
    translationKey: 'yield_weather.common.sri_gemunu',
    description: 'High-yielding Sri Lankan cinnamon variety with excellent commercial value',
    scientificName: 'Cinnamomum zeylanicum var. Sri Gemunu',
    characteristics: 'Superior bark quality, good peeling characteristics, high oil content'
  },
  {
    value: 'Sri Wijaya',
    label: 'Sri Wijaya', // Will be translated in UI
    translationKey: 'yield_weather.common.sri_wijaya',
    description: 'Premium Sri Lankan variety known for superior quality and aroma',
    scientificName: 'Cinnamomum zeylanicum var. Sri Wijaya',
    characteristics: 'Excellent flavor profile, fine bark texture, disease resistant'
  }
] as const;

// Extract just the values for easy usage
export const CINNAMON_VARIETY_VALUES = CINNAMON_VARIETIES.map(variety => variety.value);

// Default variety (most common in dataset)
export const DEFAULT_CINNAMON_VARIETY = 'Sri Gemunu';

// Variety selection options for dropdowns
export const CINNAMON_VARIETY_OPTIONS = CINNAMON_VARIETIES.map(variety => ({
  label: variety.label,
  value: variety.value,
  description: variety.description
}));